### PR TITLE
[Merged by Bors] - Remove Anyhow::Result in system_piping.rs example

### PR DESCRIPTION
**Pull request by foxzool**
# Objective

Fixes: #7610 

## Solution

- Return a specific error type

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Thanks :) You can remove the change log and migration guide sections from your PR description completely.

File: examples/ecs/system_piping.rs [hunk 1]
```diff
@@ -31,20 +31,20 @@ struct Message(String);
 struct OptionalWarning(Result<(), String>);
 
 // This system produces a Result<usize> output by trying to parse the Message resource.
-fn parse_message_system(message: Res<Message>) -> Result<usize> {
+fn parse_message_system(message: Res<Message>) -> Result<usize, ParseIntError> {
     Ok(message.parse::<usize>()?)
```
Comment (mockersf) [inline examples/ecs/system_piping.rs:35]:
```suggestion
    message.parse::<usize>()
```

**Comment by alice-i-cecile** [comment]
Title suggestion:

> Remove Anyhow::Result in system_piping.rs example

**Review by devil-ira — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@foxzool once you reformat your code (as CI suggests) I'm ready to merge this.

**Comment by foxzool** [comment]
@alice-i-cecile CI passed

**Comment by alice-i-cecile** [comment]
Solid work :)

bors r+

**Pull request closed without merging**